src/State.flix

```diff
@@ -0,0 +1,40 @@
+namespace Flixball {
+    use Flixball/Core.AiState;
+    use Flixball/Core.AiState.{Value, Collection};
+
+    pub class State[a] {
+        pub def toState(x: a): AiState
+        pub def fromState(x: AiState): Option[a]
+    }
+
+    pub class MutState[m: Region -> Type] {
+        pub def toState(x: m[r]): AiState \ r
+        pub def fromState(r: Region[r], x: AiState): Option[m[r]] \ r
+    }
+
+    instance State[Bool] {
+        pub def toState(x: Bool): AiState = Value("${x}")
+
+        pub def fromState(x: AiState): Option[Bool] = match x {
+            case Value("true") => Some(true)
+            case Value("false") => Some(false)
+            case _ => None
+        }
+    }
+
+    instance State[Int64] {
+        pub def toState(x: Int64): AiState = Value("${x}")
```

Comment:
looks good to me. I think if we were very concerned about correctness then I'm not sure that it is guaranteed that `fromString(toString(x)) == x` is always true. I don't think its a law of the type classes but I'm not sure

Reply:
That's a good point. I think as long as we don't rely on it _in general_ we should be ok. It happens to hold for `Int64` and `Bool`, so we can use it for those specifically.

Reply:
yeah. We could also use the bool instance of `FromString` here just to remove a couple strings

Reply:
yeah. We could also use the bool instance of `FromString` here just to remove a couple strings